Tests/PointFreeTests/HomeTests.swift

```diff
@@ -11,19 +11,86 @@ import WebKit
 #endif
 
 class HomeTests: TestCase {
-  func testHomepage() {
+  override func setUp() {
+    super.setUp()
+
+    let eps = [
+      ep10,
+      ep2,
+      ep1,
+      introduction,
+      ]
+      .suffix(4)
+      .map(\.image .~ "")
+
+    let env: (Environment) -> Environment =
+      (\.database .~ .mock)
+      <> (\.episodes .~ unzurry(eps))
+
+    AppEnvironment.push(env)
+  }
+
+  override func tearDown() {
+    super.tearDown()
+    AppEnvironment.pop()
+  }
+
+  func testHomepage_LoggedOut() {
     let conn = connection(from: request(to: .home))
     let result = conn |> siteMiddleware
 
     assertSnapshot(matching: result.perform())
 
     #if !os(Linux)
     if #available(OSX 10.13, *), ProcessInfo.processInfo.environment["CIRCLECI"] == nil {
-      let webView = WKWebView(frame: .init(x: 0, y: 0, width: 1080, height: 1600))
+      let webView = WKWebView(frame: .init(x: 0, y: 0, width: 1080, height: 3000))
       webView.loadHTMLString(String(data: result.perform().data, encoding: .utf8)!, baseURL: nil)
       assertSnapshot(matching: webView, named: "desktop")
+
       webView.frame.size.width = 400
-      assertSnapshot(matching: webView, named: "mobile")
+      webView.frame.size.height = 3500
+
+      let render = XCTestExpectation()
+      DispatchQueue.main.async {
+        assertSnapshot(matching: webView, named: "mobile")
+        render.fulfill()
+      }
+      guard XCTWaiter.wait(for: [render], timeout: 2.0) == .completed else {
```

Comment:
I think you can do `waitForExpectations` and save a coupla lines.

Reply:
nice, will do